Purge stdenv.cross

Ericson2314 opened the pull request:
###### Motivation for this change

`stdenv.cross` is vague: better to use `buildPlatform`, `hostPlatform` or `targetPlatform`. This is a rather tedious task so anyone that sees this, feel free to push to this PR (if you are a contributor) or send a PR against the PR (if you are not).

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @bjornfor @matthewbauer @Dridus

Ericson2314 committed:
libxml2: Don't use stdenv.cross

Ericson2314 committed:
libxslt: Don't use stdenv.cross

Ericson2314 committed:
ImageMagick: Don't use stdenv.cross

I'm guessing the salient aspect here is not cross compiling itself, but
just whether the host platform is MinGW, so I simplified the logic
accordingly

Ericson2314 committed:
fbida: cc-wrapper can be relied on to export these env vars

Ericson2314 committed:
offrss: cc-wrapper can be relied on to export this env var

Ericson2314 committed:
fossil: Don't use stdenv.cross

Ericson2314 committed:
omxplayer: Don't use stdenv.cross

Ericson2314 committed:
make-initrd: Don't use stdenv.cross

Ericson2314 committed:
guile: Don't use stdenv.cross

Ericson2314 committed:
lua 5: Don't use stdenv.cross, and use less make flags

cc-wrapper will define environment variables matching those now

Ericson2314 committed:
SDL: Don't use stdenv.cross

Mic92 commented on pkgs/development/libraries/libxml2/default.nix:
```diff
@@ -1,7 +1,9 @@
+, buildPlatform, hostPlatform
+, pythonSupport ? buildPlatform == hostPlatform
 , icuSupport ? false, icu ? null }:
+}:
```
closes parentheses twice.

Ericson2314 replied:
Thanks; fixed

Ericson2314 committed:
scummvm: Don't use stdenv.cross

Ericson2314 committed:
boost: Don't use stdenv.cross, and utilize new platform metadata

Ericson2314 committed:
ffmpeg: Don't use stdenv.cross

Ericson2314 committed:
libvpx: Don't use stdenv.cross

Ericson2314 committed:
glew: Rely on cc-wrapper can to export these env vars

Ericson2314 committed:
cc-wrapper: Also export env var for windres, when it is present

Ericson2314 committed:
openssl: cc-wrapper can be relied on to export these env vars

Ericson2314 committed:
vim: cc-wrapper can be relied on to export this env var

Ericson2314 committed:
qt 4.8: Don't use stdenv.cross

Ericson2314 committed:
WIP: optipng: cc-wrapper can be relied on to export these env vars

Ericson2314 committed:
zlib: cc-wrapper can be relied on to export this env var

hSloan committed:
gcc-wrapper-old: Don't use stdenv.cross

Take the dynamic linker logic for cc-wrapper for now

hSloan committed:
SDL: Don't use stdenv ? cross

Simplify and finess logic a bit in a way that doesn't affect the
default native build, too.

hSloan committed:
boehm-gc: Don't use stdenv ? cross

hSloan committed:
boost: Don't use stdenv ? cross

hSloan committed:
fontconfig 2.10: Don't use stdenv.cross

hSloan committed:
fontconfig default: Don't use stdenv.cross

hSloan committed:
freetype default: Don't use stdenv.cross

hSloan committed:
glew 1.10: Don't use stdenv ? cross

hSloan committed:
glew default: Don't use stdenv ? cross

hSloan committed:
libav: Don't use stdenv.cross

hSloan committed:
libffi: Don't use stdenv ? cross

hSloan committed:
libiconv: Don't use stdenv ? cross

hSloan committed:
libjpeg-turbo: Don't use stdenv.cross

hSloan committed:
libmsgpack generic: Don't use stdenv.cross

hSloan committed:
libpng 12: Don't use stdenv ? cross

hSloan committed:
libpng default: Don't use stdenv.cross

hSloan committed:
libssh2: Don't use stdenv.cross

hSloan committed:
newt: Don't use stdenv.cross

hSloan committed:
nlohmann_json: Don't use stdenv.cross

hSloan committed:
poppler: Don't use stdenv.cross

hSloan committed:
readline 6.2: Don't use stdenv ? cross

hSloan committed:
readline 6.3: Don't use stdenv ? cross

hSloan committed:
readline 7.0: Don't use stdenv ? cross

hSloan committed:
cmake 2.8: Don't use stdenv ? cross

hSloan committed:
libtool2: Don't use stdenv ? cross

hSloan committed:
uboot: Don't use stdenv.cross

hSloan committed:
kernel-headers: Don't use stdenv.cross

hSloan committed:
kernel generic: Don't use stdenv.cross

hSloan committed:
kernel manual-config: Don't use stdenv.cross

hSloan committed:
kernel perf: Don't use stdenv.cross

hSloan committed:
klibc: Don't use crossAttrs

hSloan committed:
mdadm: Don't use stdenv.cross

hSloan committed:
mingetty: Rely on cc-wrapper to export this env var

hSloan committed:
propcps: Rely on cc-wrapper to export this env var

hSloan committed:
shadow: Don't use stdenv ? cross

hSloan committed:
sysvinit: Rely on cc-wrapper to export this env var

hSloan committed:
jom: Don't use stdenc.cross

hSloan committed:
wxMSW-2.8: Don't use stdenv ? cross

hSloan committed:
mtdutils: Rely on cc-wrapper to export this env var

hSloan committed:
tcpdump: Don't use stdenv.cross

hSloan committed:
nix: Don't use stdenv.cross

hSloan committed:
tm: Rely on cc-wrapper to export this env var

hSloan committed:
ts: Rely on cc-wrapper to export this env var

hSloan committed:
html-tidy: Don't use stdenv.cross

hSloan committed:
replace: Rely on cc-wrapper to export env var

Ericson2314 commented:
@bjornfor that's my fault with the sed before the manual fix up. I hope there's a way to filter-branch to fix that....

bjornfor commented:
@Ericson2314. No problem. I think you can script the git filter-branch thing.

hSloan committed:
top-level: stdenv.cross vanquished
